theano/tensor/nlinalg.py

```diff
@@ -179,6 +185,10 @@ class ExtractDiag(Op):
     __props__ = ("view",)
 
     def __init__(self, view=False):
+        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.ExtractDiag"
```

Comment:
The opt will build a graph such the the input variable is an
CudaNdarrayVariable, so at run time, it will receive an CudaNdarray
instance. The make_node of ExtractDiag, have this at the start:

```
    def make_node(self, _x):
        if not isinstance(_x, theano.Variable):
            x = as_tensor_variable(_x)
        else:
            x = _x
```

So it don't convert a CudaNdarrayVariable to a TensorVarible.

On Wed, Mar 9, 2016 at 9:38 AM, Xavier Bouthillier <notifications@github.com
> wrote:

> In theano/tensor/nlinalg.py
> <https://github.com/Theano/Theano/pull/4169#discussion_r55526614>:
>
> > @@ -179,6 +185,10 @@ class ExtractDiag(Op):
> >      __props__ = ("view",)
> >
> >      def __init__(self, view=False):
> > +        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.ExtractDiag"
>
> Really? [image: :open_mouth:] How could it be? Any access to x like x[0]
> or x[:, 0] returns a CudaNDarray? It only work if there is no arithmetic on
> x right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4169/files#r55526614>.
>